examples/interaction_box_points.py

```diff
@@ -1,51 +0,0 @@
-"""
-Interaction box points
-======================
-
-Demonstrate interaction box on points layer
```

Comment:
No, this is the main breaking change of this PR. I probably wasn't clear, but that's what I meant here:

> selection_box is using the same overlay machinery from transform_box, but it's intended to replace the custom selection logic that Points and other layers implement with a unified system, while also allowing to transform the selected object (like interaction_box was meant to do before, at least for points). This is not yet implemented, but I think it should happen in a followup PR.

Basically, on main, `interaction_box` could be used to transform points (not the layer, but individual or groups of points within the layer!) if you hook it up in a certain way (all is done manually in the example, none of it is really napari core). My changes make this impossible, which is why this example is broken. However, the newly added `selection_box` (which is only hinted at, not really implemented) will take the place of this functionality, as well as unifying the selection logic we have in points and hopefully providing the same thing for vectors, shapes, etc.

I think this is a very niche use, and probably only @jojoelfe was using it, which is why I tagged them. I'm happy to extend this PR to include that if we believe that the breaking change is a problem. Otherwise, I suggest to do it in a followup.

Reply:
I think another reason to justify this breaking change is that the interaction box functionality was introduced fairly recently.